File: apps/mobile/src/models/Farmer.js
import { Model } from '@nozbe/watermelondb'
import { field, date, readonly } from '@nozbe/watermelondb/decorators'

export default class Farmer extends Model {
  static table = 'farmers'

  @field('server_id')    serverId     // server UUID — null until first sync
  @field('phone')        phone
  @field('name')         name
  @field('kyc_status')   kycStatus    // PENDING | VERIFIED | REJECTED
  @field('fpo_id')       fpoId
  @field('is_deleted')   isDeleted

  @readonly @date('created_at') createdAt
  @date('updated_at')           updatedAt
  @date('last_synced_at')       lastSyncedAt

  get isVerified() { return this.kycStatus === 'VERIFIED' }
  get isPending()  { return this.kycStatus === 'PENDING' }
  get canReceivePayout() { return this.isVerified && !this.isDeleted }
}

File: apps/backend/domain/farmer.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type KYCStatus string

const (
	KYCPending  KYCStatus = "PENDING"
	KYCVerified KYCStatus = "VERIFIED"
	KYCRejected KYCStatus = "REJECTED"
)

type Farmer struct {
	ID           uuid.UUID
	Phone        string
	Name         string
	AadhaarHash  string // SHA-256 of Aadhaar — never raw
	KYCStatus    KYCStatus
	FPOID        *uuid.UUID
	CreatedAt    time.Time
	UpdatedAt    time.Time
	DeletedAt    *time.Time
	LastSyncedAt *time.Time
}

func (f *Farmer) IsActive() bool { return f.DeletedAt == nil }

func (f *Farmer) CanReceivePayout() error {
	if !f.IsActive() {
		return ErrValidation("farmer account is deactivated")
	}
	if f.KYCStatus != KYCVerified {
		return ErrValidation("farmer KYC is not verified — cannot receive payout")
	}
	return nil
}
